wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java

```diff
@@ -1388,27 +1388,35 @@ public WorkunitInfo createWUFromECL(WorkunitInfo wu) throws Exception
             WUUpdateResponse wuUpdateResponse = wsWorkunitsServiceSoapProxy.WUCreateAndUpdate(wucreateparameters);
 
             ArrayOfEspException exceptions = wuUpdateResponse.getExceptions();
-            if (exceptions == null)
-            {
-                createdWU = new WorkunitInfo(wuUpdateResponse.getWorkunit());
-                createdWU.setMaxMonitorMillis(wu.getMaxMonitorMillis());
-                createdWU.setCluster(wu.getCluster());
-                createdWU.setJobname(wu.getJobname());
-                createdWU.setSleepMillis(wu.getSleepMillis());
-                createdWU.setOriginalEclWatchUrl(getEclWatchUrl());
-
-            }
-            else
+            if (exceptions != null)
+        	    throwWsWUExceptions(exceptions, "Error compiling ECL query");
+            if (wuUpdateResponse.getWorkunit().getStateID() != 1) // not immediately in compiled state
             {
-                throwWsWUExceptions(exceptions, "Error compiling ECL query");
+                WUWaitCompiled wuwaitcompiledparameters = new WUWaitCompiled();
+                wuwaitcompiledparameters.setWait(10); // seconds
```

Comment:
these timeouts are always so tough to get right, does it make sense to let the caller pass in this value?

Reply:
Does it make sense to use wu.maxMonitorMillis? or the defaultWaitTime variable?